src/crypto/clu_crypto_setup.c

```diff
@@ -0,0 +1,410 @@
+            out = optarg;
+            outCheck = 1;
+            break;
+
+        case WOLFCLU_INKEY:
+            if (optarg == NULL) {
+                WOLFCLU_LOG(WOLFCLU_L0, "no key passed in..");
+                wolfCLU_freeBins(pwdKey, iv, key, NULL, NULL);
+                return WOLFCLU_FATAL_ERROR;
+            }
+
+            /* 2 characters = 1 byte. 1 byte = 8 bits
+             */
+            numBits = (word32)(XSTRLEN(optarg) * 4);
+            /* Key for encryption */
+            if ((int)numBits != keySize) {
+                WOLFCLU_LOG(WOLFCLU_L0,
+                        "Length of key provided was: %d.", numBits);
+                WOLFCLU_LOG(WOLFCLU_L0,
+                        "Length of key expected was: %d.", keySize);
+                WOLFCLU_LOG(WOLFCLU_L0,
+                        "Invalid Key. Must match algorithm key size.");
+                wolfCLU_freeBins(pwdKey, iv, key, NULL, NULL);
+                return WOLFCLU_FATAL_ERROR;
+            }
+            else {
+                char* keyString;
+
+                keyString = (char*)XMALLOC(XSTRLEN(optarg), HEAP_HINT,
+                        DYNAMIC_TYPE_TMP_BUFFER);
+                if (keyString == NULL) {
+                    wolfCLU_freeBins(pwdKey, iv, key, NULL, NULL);
+                    return MEMORY_E;
+                }
+
+                XSTRNCPY(keyString, optarg, XSTRLEN(optarg));
+                ret = wolfCLU_hexToBin(keyString, &key, &numBits,
+                                       NULL, NULL, NULL,
+                                       NULL, NULL, NULL,
+                                       NULL, NULL, NULL);
+                XFREE(keyString, HEAP_HINT, DYNAMIC_TYPE_TMP_BUFFER);
+                if (ret != 0) {
+                    WOLFCLU_LOG(WOLFCLU_L0,
+                            "failed during conversion of Key, ret = %d", ret);
+                    wolfCLU_freeBins(pwdKey, iv, key, NULL, NULL);
+                    return WOLFCLU_FATAL_ERROR;
+                }
+                keyCheck = 1;
+                keyType = 2;
+            }
+            break;
+
+        case WOLFCLU_SIGFILE:
+            break;
+
+        case WOLFCLU_DECRYPT:
+            encCheck = 0;
+            decCheck = 1;
+            break;
+
+        case WOLFCLU_DEBUG:
+            verbose = 1;
+            break;
+
+        case WOLFCLU_MD:
+            hashType = wolfSSL_EVP_get_digestbyname(optarg);
+            if (hashType == NULL) {
+                WOLFCLU_LOG(WOLFCLU_L0, "invalid digest name");
+                return WOLFCLU_FATAL_ERROR;
+            }
+            break;
+
+        case ':':
+        case '?':
+            break;
+
+        default:
+            /* do nothing. */
+            (void)ret;
+        }
+    }
+
+    if (pwdKeyChk == 0 && keyCheck == 0) {
+        if (decCheck == 1) {
+            WOLFCLU_LOG(WOLFCLU_L0, "\nDECRYPT ERROR:");
+            WOLFCLU_LOG(WOLFCLU_L0, "no key set or passphrase");
```

Comment:
If I understand the error message correctly, better wording would be "no key or passphrase set."